test/common/config/http_subscription_impl_test.cc

```diff
@@ -42,6 +42,26 @@ TEST_F(HttpSubscriptionImplTest, BadJsonRecovery) {
   verifyStats(3, 1, 0, 1, 7148434200721666028);
 }
 
+TEST_F(HttpSubscriptionImplTest, ConfigNotModified) {
```

Comment:
@cetanu I think you were almost there before, i.e. 2 commits before, but without expecting failure:

```c++
TEST_F(HttpSubscriptionImplTest, ConfigNotModified) {
  startSubscription({"cluster0", "cluster1"});

  verifyStats(1, 0, 0, 0, 0);
  timerTick();
  verifyStats(2, 0, 0, 0, 0);

  // accept and modify.
  deliverConfigUpdate({"cluster0", "cluster1"}, "0", true, true, "200");
  verifyStats(3, 1, 0, 0, 7148434200721666028);

  // accept and does not modify.
  deliverConfigUpdate({"cluster0", "cluster1"}, "0", true, false, "304");
  verifyStats(4, 1, 0, 0, 7148434200721666028);
}

```